common/storage/sql/index/folder-size-cache.go

```diff
@@ -141,80 +161,82 @@ func (dao *FolderSizeCacheSQL) GetNodeTree(ctx context.Context, path *tree.MPath
-	dao.invalidateMPathHierarchy(node.GetMPath(), -1)
+	dao.invalidateMPathHierarchy(ctx, node.GetMPath(), -1)
 	return dao.DAO.UpdateNode(ctx, node)
 }
 
 // DelNode removes a node from the tree
 func (dao *FolderSizeCacheSQL) DelNode(ctx context.Context, node tree.ITreeNode) error {
-	dao.invalidateMPathHierarchy(node.GetMPath(), -1)
+	dao.invalidateMPathHierarchy(ctx, node.GetMPath(), -1)
 	return dao.DAO.DelNode(ctx, node)
 }
 
 // MoveNodeTree move all the nodes belonging to a tree by calculating the new mpathes
 func (dao *FolderSizeCacheSQL) MoveNodeTree(ctx context.Context, nodeFrom tree.ITreeNode, nodeTo tree.ITreeNode) error {
 	root := nodeTo.GetMPath().CommonRoot(nodeFrom.GetMPath())
 
-	dao.invalidateMPathHierarchy(nodeTo.GetMPath(), root.Length())
-	dao.invalidateMPathHierarchy(nodeFrom.GetMPath(), root.Length())
+	dao.invalidateMPathHierarchy(ctx, nodeTo.GetMPath(), root.Length())
+	dao.invalidateMPathHierarchy(ctx, nodeFrom.GetMPath(), root.Length())
 
 	return dao.DAO.MoveNodeTree(ctx, nodeFrom, nodeTo)
 }
 
-func (dao *FolderSizeCacheSQL) invalidateMPathHierarchy(mpath *tree.MPath, level int) {
+func (dao *FolderSizeCacheSQL) invalidateMPathHierarchy(ctx context.Context, mpath *tree.MPath, level int) {
+	ca, err := cache_helper.ResolveCache(ctx, common.CacheTypeShared, folderSizeCacheConfig)
+	if err != nil {
+		return
+	}
 
 	parents := mpath.Parents()
 	if level > -1 {
 		parents = mpath.Parents()[level:]
 	}
 
-	folderSizeLock.Lock()
-	delete(folderSizeCache, mpath.ToString())
-	folderSizeLock.Unlock()
-
+	_ = ca.Delete(mpath.ToString())
 	for _, p := range parents {
-		folderSizeLock.Lock()
-		delete(folderSizeCache, p.ToString())
-		folderSizeLock.Unlock()
+		_ = ca.Delete(p.ToString())
 	}
 }
 
 // Compute sizes from children files - Does not handle lock, should be
 // used by other functions handling lock
-func (dao *FolderSizeCacheSQL) folderSize(ctx context.Context, node tree.ITreeNode) {
-
+func (dao *FolderSizeCacheSQL) folderSize(ctx context.Context, node tree.ITreeNode) (err error) {
 	mpath := node.GetMPath().ToString()
-
-	folderSizeLock.RLock()
-	size, ok := folderSizeCache[mpath]
-	folderSizeLock.RUnlock()
-
-	if ok {
-		node.GetNode().SetSize(int64(size))
+	ca, err := cache_helper.ResolveCache(ctx, common.CacheTypeShared, folderSizeCacheConfig)
+	if err != nil {
 		return
 	}
 
+	if data, ok := ca.GetBytes(mpath); ok && data != nil {
+		if sizeInt, err := strconv.ParseInt(string(data), 10, 64); err == nil && sizeInt > 0 {
+			node.GetNode().SetSize(sizeInt)
+		}
+	}
+
 	size, err := dao.GetNodeChildrenSize(ctx, node.GetMPath())
 	if err != nil {
 		return
 	}
 
 	node.GetNode().SetSize(int64(size))
 
-	folderSizeLock.Lock()
-	folderSizeCache[mpath] = size
-	folderSizeLock.Unlock()
+	// using a string representation for byte size since redis does not support int64.
+	err = ca.Set(mpath, []byte(strconv.FormatInt(int64(size), 10)))
+	if err != nil {
+		return err
+	}
+	return nil
```

Comment:
Not sure this check is necessary
```suggestion
	return err
```

Reply:
I removed the err return, just return